spec/ameba/rule/lint/unused_literal_spec.cr

```diff
@@ -0,0 +1,267 @@
+        {goodnight: moon}
+      # ^^^^^^^^^^^^^^^^^ error: Literal value is not used
+        {1, 2, 3}
+      # ^^^^^^^^^ error: Literal value is not used
+        <<-HEREDOC
+      # ^^^^^^^^^^ error: Literal value is not used
+          this is a heredoc
+          HEREDOC
+      end
+      CRYSTAL
+    end
+
+    it "fails if literals are unused in case" do
+      expect_issue subject, <<-CRYSTAL
+        thing = case object
+                when Type1
+                  true
+                # ^^^^ error: Literal value is not used
+                  "meow"
+                else
+                  if begin
+                      "huh?"
+                    # ^^^^^^ error: Literal value is not used
+                      1 > 0
+                    end
+                    1234_f32
+                  # ^^^^^^^^ error: Literal value is not used
+                  end
+
+                  "woof"
+                end
+        CRYSTAL
+    end
+
+    it "fails if literals are unused in class bodies" do
+      expect_issue subject, <<-CRYSTAL
+        class MyClass
+          case object
+          when Type1
+            true
+          # ^^^^ error: Literal value is not used
+            "meow"
+          # ^^^^^^ error: Literal value is not used
+          else
+            if begin
+                "huh?"
+              # ^^^^^^ error: Literal value is not used
+                1 > 0
+              end
+              1234_f32
+            # ^^^^^^^^ error: Literal value is not used
+            end
+
+            "woof"
+          # ^^^^^^ error: Literal value is not used
+          end
+        end
+        CRYSTAL
+    end
+
+    it "fails if unused literals in rescue/ensure/else block" do
+      expect_issue subject, <<-CRYSTAL
+      a = begin
+            1234
+          # ^^^^ error: Literal value is not used
+            1234_f32
+          rescue ASDF
+            "hello world"
+          # ^^^^^^^^^^^^^ error: Literal value is not used
+            "interp \#{string}"
+          rescue QWERTY
+            [1, 2, 3, 4, 5]
+          # ^^^^^^^^^^^^^^^ error: Literal value is not used
+            {"hello" => "world"}
+          else
+            '\t'
+          # ^^^ error: Literal value is not used
+            <<-HEREDOC
+          # ^^^^^^^^^^ error: Literal value is not used
+              this is a heredoc
+              HEREDOC
+            1..2
+          ensure
+            {goodnight: moon}
+          # ^^^^^^^^^^^^^^^^^ error: Literal value is not used
+            {1, 2, 3}
+          end
+      CRYSTAL
+    end
+
+    # Locations for Regex literals were added in Crystal v1.15.0
+    {% if compare_versions(Crystal::VERSION, "1.15.0") >= 0 %}
+      it "fails if a regex literal is unused" do
+        expect_issue subject, <<-CRYSTAL
+            a = /hello world/
+            /goodnight moon/
+          # ^^^^^^^^^^^^^^^^ error: Literal value is not used
+          CRYSTAL
+      end
+    {% end %}
```

Comment:
do we want to add a test for the opposite situation to ensure it works as expected on Crystal < 1.15? Might be subjective.

Reply:
That works for me

Reply:
Found a small bug because of this, thanks for the catch